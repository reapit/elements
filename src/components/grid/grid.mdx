import { Story, Canvas, Controls, Meta } from '@storybook/blocks'
import { GridResponsive, ColResponsive } from './grid-responsive'
import { RenderHtmlMarkup } from '../../storybook/render-html-markup'
import * as GridStories from './grid.stories'

<Meta of={GridStories} />

# Grid

The Grid leverages CSS Grid to provide a responsive layout for your applications.

## Basic Usage

The basic grid should fulil majority of requirements for most users and is based on Foundations Styleguide responsive breakpoints.

There are two core components `Grid` and `Col`. In all cases `Col` will span 4 columns of grid, whilst number of grid columns increases as screen size increases.

The table below details breakpoints from styleguide against ther respective screen size and column counts when using grid.

<table>
  <thead>
    <tr>
      <td>Breakpoint Variable</td>
      <td>Window Width</td>
      <td>Column count</td>
      <td>Items Per Row</td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>isMobile</td>
      <td>&lt; 767px</td>
      <td>4</td>
      <td>1</td>
    </tr>
    <tr>
      <td>isTablet</td>
      <td>&gt;= 768 and &lt; 1024</td>
      <td>4</td>
      <td>1</td>
    </tr>
    <tr>
      <td>isDesktop</td>
      <td>&gt;= 1024 and &lt; 1440</td>
      <td>8</td>
      <td>2</td>
    </tr>
    <tr>
      <td>isWideScreen</td>
      <td>&gt;= 1440 and &lt; 1920</td>
      <td>12</td>
      <td>3</td>
    </tr>
    <tr>
      <td>isSuperWideScreen</td>
      <td>&gt;= 1920 and &lt; 2560</td>
      <td>16</td>
      <td>4</td>
    </tr>
    <tr>
      <td>is4KScreen</td>
      <td>&gt;= 2560</td>
      <td>20</td>
      <td>5</td>
    </tr>
  </tbody>
</table>

Row and column gaps are standardised in line with our style guide.

<Canvas of={GridStories.BasicUsage} />

<RenderHtmlMarkup of={GridStories.BasicUsage} />

## Basic Usage Quarter and Halves

Using the basic grid you can also use smaller columns for more granular control over your layout with `ColHalfy` and `ColQuarter` components.

<Canvas of={GridStories.BasicUsageQuarterAndHalves} />

<RenderHtmlMarkup of={GridStories.BasicUsageQuarterAndHalves} />

## Basic Usage - Split Screen Grid

For more simple layouts, you will likely need a split screen, two column layout. This can be achieved by substituting `Col` component with a `ColSplit` grid item.

Below 1024px the grid will have a single item per column, above, the screen will split into two columns.

<Canvas of={GridStories.BasicUsageSplitScreenGrid} />

<RenderHtmlMarkup of={GridStories.BasicUsageSplitScreenGrid} />

## Basic Usage - Split Screen Grid Thirds

For a sidebar and main colum view, you can use a `GridThirds` grid, with `ColSplitThird` and `ColSplitTwoThirds` grid items - this is best combined with a `hasMaxWidth` Container - see layouts for more on this.

In mobile and tablet, these columns stack then break at destop view and above.

<Canvas of={GridStories.BasicUsageSplitScreenGridThirds} />

<RenderHtmlMarkup of={GridStories.BasicUsageSplitScreenGridThirds} />

## Custom Usage

If you need more granular control over your grid, you can override base with utility classes we expose.

The grid class / component can have both it's `grid-column-gap` and `grid-row-gap` properties with modifiers classes that follow pattern below:

- `.e-col-gap{unit}`
- `.el-row-gap{unit}`

Where `{unit}` value is one of

- `0`: 0rem
- `1`: 0.25rem
- `2`: 0.5rem
- `3`: 0.75rem
- `4`: 1rem
- `6`: 1.5rem
- `8`: 2rem
- `10`: 2.5rem
- `12`: 3rem

So a class `.el-col-gap-4` will give a `grid-column-gap` property of 1rem. Note if importing TypeScript components, classes are `camelCased` eg `elColGap4`

Similarly, for `Col` component / class, you can override both `grid-column-end` and `grid-column-start` properties with following options;

- `.e-span{unit}`
- `.el-offset{unit}`

Where `{unit}` value a number `1 - 12`, mapping to number of columns `Col` should span and column it should start at on grid respectively.

The below contrived example we have overridden base behaviour by adding a custom column and row gap to grid and start and span properties to columns.

Obviously to make classes responsive, you will need some JavaScript to toggle classes as appropriate.

<Canvas of={GridStories.CustomUsage} />

<RenderHtmlMarkup of={GridStories.CustomUsage} />

## React Example - Responsive Customisation

In previous example, we saw that customisation required some JS logic to toggle classes based on media widths.

For React users, we have an addional `GridResponsive` component that under hood uses `useMediaQuery` hook to allow maximum customisation based on our 4 media query breakpoints.

Both `GridResponsive` and `ColResponsive` components accept props in following interface, mapping to the breakpoint variables we saw in the main grid component table.

<Controls component={GridResponsive} />

Essentially for `GridResponsive` we can set `grid-column-gap` and `grid-row-gap` for each of mobile, tablet, desktop and widescreen based on relevant prop.

For `ColResponsive` same applies, but for `grid-column-end` and `grid-column-start` properties.

In all cases, class is toggled as screen size changes, defaulting to base grid values if no additional class is applied.

Full example below creates a grid that doubles the number of columns versus the standard grid and increases both row and column gap incrementally.

## ColResponsive

<Controls component={ColResponsive} />

<Canvas of={GridStories.ReactExampleResponsiveCustomisation} />

<RenderHtmlMarkup of={GridStories.ReactExampleResponsiveCustomisation} />
