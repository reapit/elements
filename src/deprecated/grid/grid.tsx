import React, { FC, HTMLAttributes } from 'react'
import { cx } from '@linaria/core'
import {
  ElGrid,
  ElCol,
  ElColSplit,
  ElColHalf,
  ElColSplitThird,
  ElColSplitTwoThirds,
  ElGridThirds,
  ElColQuarter,
} from './__styles__'

/** @deprecated */
export interface GridProps extends HTMLAttributes<HTMLElement> {}

/** @deprecated */
export interface ColProps extends HTMLAttributes<HTMLElement> {}

/** @deprecated */
export const Grid: FC<GridProps> = ({ className, children, ...rest }: GridProps) => (
  <ElGrid className={cx(className)} {...rest}>
    {children}
  </ElGrid>
)

/** @deprecated */
export const Col: FC<ColProps> = ({ className, children, ...rest }: ColProps) => (
  <ElCol className={cx(className)} {...rest}>
    {children}
  </ElCol>
)

/** @deprecated */
export const ColHalf: FC<ColProps> = ({ className, children, ...rest }: ColProps) => (
  <ElColHalf className={cx(className)} {...rest}>
    {children}
  </ElColHalf>
)

/** @deprecated */
export const ColQuarter: FC<ColProps> = ({ className, children, ...rest }: ColProps) => (
  <ElColQuarter className={cx(className)} {...rest}>
    {children}
  </ElColQuarter>
)

/** @deprecated */
export const ColSplit: FC<ColProps> = ({ className, children, ...rest }: ColProps) => (
  <ElColSplit className={cx(className)} {...rest}>
    {children}
  </ElColSplit>
)

/** @deprecated */
export const GridThirds: FC<GridProps> = ({ className, children, ...rest }: GridProps) => (
  <ElGridThirds className={cx(className)} {...rest}>
    {children}
  </ElGridThirds>
)

/** @deprecated */
export const ColSplitThird: FC<ColProps> = ({ className, children, ...rest }: ColProps) => (
  <ElColSplitThird className={cx(className)} {...rest}>
    {children}
  </ElColSplitThird>
)

/** @deprecated */
export const ColSplitTwoThirds: FC<ColProps> = ({ className, children, ...rest }: ColProps) => (
  <ElColSplitTwoThirds className={cx(className)} {...rest}>
    {children}
  </ElColSplitTwoThirds>
)
