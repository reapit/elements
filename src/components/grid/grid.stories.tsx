import { GridDemoBlock } from '../../storybook/demo-block'
import { Grid, Col, ColHalf, ColQuarter, ColSplit, ColSplitThird, ColSplitTwoThirds, GridThirds } from './grid'
import { GridResponsive, ColResponsive } from './grid-responsive'
import { MediaStateProvider } from '../../hooks/use-media-query'
import { cx } from '@linaria/core'
import { elColGap2, elRowGap3, elSpan12, elSpan8, elSpan4, elOffset4, elOffset8 } from './__styles__/units'

export default {
  title: 'Grid',
}

export const BasicUsage = {
  render: ({}) => (
    <Grid>
      <Col>
        <GridDemoBlock />
      </Col>
      <Col>
        <GridDemoBlock />
      </Col>
      <Col>
        <GridDemoBlock />
      </Col>
      <Col>
        <GridDemoBlock />
      </Col>
      <Col>
        <GridDemoBlock />
      </Col>
    </Grid>
  ),
}

export const BasicUsageQuarterAndHalves = {
  render: ({}) => (
    <Grid>
      <ColHalf>
        <GridDemoBlock />
      </ColHalf>
      <ColHalf>
        <GridDemoBlock />
      </ColHalf>
      <ColHalf>
        <GridDemoBlock />
      </ColHalf>
      <ColHalf>
        <GridDemoBlock />
      </ColHalf>
      <ColQuarter>
        <GridDemoBlock />
      </ColQuarter>
      <ColQuarter>
        <GridDemoBlock />
      </ColQuarter>
      <ColQuarter>
        <GridDemoBlock />
      </ColQuarter>
      <ColQuarter>
        <GridDemoBlock />
      </ColQuarter>
      <ColQuarter>
        <GridDemoBlock />
      </ColQuarter>
      <ColQuarter>
        <GridDemoBlock />
      </ColQuarter>
    </Grid>
  ),
  name: 'Basic Usage Quarter and Halves',
}

export const BasicUsageSplitScreenGrid = {
  render: ({}) => (
    <Grid>
      <ColSplit>
        <GridDemoBlock />
      </ColSplit>
      <ColSplit>
        <GridDemoBlock />
      </ColSplit>
    </Grid>
  ),
  name: 'Basic Usage - Split Screen Grid',
}

export const BasicUsageSplitScreenGridThirds = {
  render: ({}) => (
    <GridThirds>
      <ColSplitThird>
        <GridDemoBlock />
      </ColSplitThird>
      <ColSplitTwoThirds>
        <GridDemoBlock />
      </ColSplitTwoThirds>
    </GridThirds>
  ),
  name: 'Basic Usage - Split Screen Grid Thirds',
}

export const CustomUsage = {
  render: ({}) => (
    <Grid className={cx(elColGap2, elRowGap3)}>
      <Col className={cx(elSpan12)}>
        <GridDemoBlock />
      </Col>
      <Col className={cx(elSpan8, elOffset4)}>
        <GridDemoBlock />
      </Col>
      <Col className={cx(elSpan4, elOffset4)}>
        <GridDemoBlock />
      </Col>
      <Col className={cx(elSpan4, elOffset8)}>
        <GridDemoBlock />
      </Col>
    </Grid>
  ),
}

export const ReactExampleResponsiveCustomisation = {
  render: ({}) => (
    <MediaStateProvider>
      <GridResponsive
        colGapMobile={1}
        colGapTablet={2}
        colGapDesktop={3}
        colGapWideScreen={4}
        colGapSuperWideScreen={6}
        colGap4KScreen={8}
        rowGapMobile={1}
        rowGapTablet={2}
        rowGapDesktop={3}
        rowGapWideScreen={4}
        rowGapSuperWideScreen={6}
        rowGap4KScreen={8}
      >
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
        <ColResponsive
          spanMobile={2}
          spanTablet={2}
          spanDesktop={2}
          spanWideScreen={2}
          spanSuperWideScreen={2}
          span4KScreen={2}
        >
          <GridDemoBlock />
        </ColResponsive>
      </GridResponsive>
    </MediaStateProvider>
  ),
  name: 'React Example - Responsive Customisation',
}
