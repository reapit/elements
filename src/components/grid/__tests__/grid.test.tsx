import { render } from '@testing-library/react'
import { Grid, Col, ColSplit, GridThirds, ColSplitThird, ColSplitTwoThirds, ColHalf, ColQuarter } from '../grid'

describe('Grid', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <Grid>
        <p>I am child</p>
      </Grid>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('Col', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <Col>
        <p>I am child</p>
      </Col>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('ColHalf', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <ColHalf>
        <p>I am child</p>
      </ColHalf>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('ColQuarter', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <ColQuarter>
        <p>I am child</p>
      </ColQuarter>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('ColSplit', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <ColSplit>
        <p>I am child</p>
      </ColSplit>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('GridThirds', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <GridThirds>
        <p>I am child</p>
      </GridThirds>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('ColSplitThird', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <ColSplitThird>
        <p>I am child</p>
      </ColSplitThird>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})

describe('ColSplitTwoThirds', () => {
  it('should match a snapshot and render children', () => {
    const wrapper = render(
      <ColSplitTwoThirds>
        <p>I am child</p>
      </ColSplitTwoThirds>,
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
