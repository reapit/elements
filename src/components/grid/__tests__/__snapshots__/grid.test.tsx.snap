// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Col > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-1 el-col"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-1 el-col"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColHalf > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-2 el-col-half"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-2 el-col-half"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColQuarter > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-3 el-col-quarter"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-3 el-col-quarter"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColSplit > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-4 el-col-split"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-4 el-col-split"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColSplitThird > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-6 el-col-split-third"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-6 el-col-split-third"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColSplitTwoThirds > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-7 el-col-split-two-thirds"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-7 el-col-split-two-thirds"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Grid > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-0 el-grid"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-0 el-grid"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`GridThirds > should match a snapshot and render children 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-5 el-grid-thirds"
      >
        <p>
          I am child
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-5 el-grid-thirds"
    >
      <p>
        I am child
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
